libtransmission/peer-msgs.cc

```diff
@@ -1250,7 +1244,7 @@ void tr_peerMsgsImpl::parse_ltep_handshake(MessageReader& payload)
     // look for upload_only (BEP 21)
     if (auto upload_only = int64_t{}; tr_variantDictFindInt(&*var, TR_KEY_upload_only, &upload_only))
     {
-        pex.flags |= ADDED_F_SEED_FLAG;
+        peer_info->set_seed();
```

Comment:
why not peer_info->set_upload_only() ?

or both?

Note: qBittorrent has weird behavior where it sends upload_only in LTEP but then follows up with fext-have-none message.  Makes no sense, but thats in the wild:

```
[2024-06-24 17:19:17.858] trc handshake 72.##.##.##:45263 peer-id is 'qBittorrent 4.6.3' ... isIncoming is false (handshake.cc:338)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 on_handshake_done: result.is_connected true, swarm is_running true (peer-mgr.cc:1338)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: sending an ltep handshake (peer-msgs.cc:1081)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: sending 'ltep' 0 [] (peer-msgs.cc:850)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: sending 'fext-have-all' (peer-msgs.cc:850)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: read 160 payload bytes; 0 left to go (peer-msgs.cc:1770)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: got peer msg 'ltep' (20) with payload len 160 (peer-msgs.cc:1356)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: Got a BtPeerMsgs::Ltep (peer-msgs.cc:1583)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: got ltep handshake (peer-msgs.cc:902)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: here is the base64-encoded handshake: ...
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent 4.6.3]: peer is upload_only (peer-msgs.cc:1255)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent/4.6.3]: read 0 payload bytes; 0 left to go (peer-msgs.cc:1770)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent/4.6.3]: got peer msg 'fext-have-none' (15) with payload len 0 (peer-msgs.cc:1356)
[2024-06-24 17:19:17.858] trc TORRENTNAME 72.##.##.##:45263 [qBittorrent/4.6.3]: Got a BtPeerMsgs::FextHaveNone (peer-msgs.cc:1546)
[2024-06-24 17:19:17.858] trc 72.##.##.##:45263 enabling ready-to-write polling (peer-io.cc:525)
[2024-06-24 17:19:17.858] trc 72.##.##.##:45263 libevent says this peer socket is ready for writing (peer-io.cc:357)
[2024-06-24 17:19:17.858] trc 72.##.##.##:45263 libevent says this peer socket is ready for reading (peer-io.cc:488)
[2024-06-24 17:19:17.858] trc 72.##.##.##:45263 try_read err: n_read:0 errno:107 (Transport endpoint is not connected) (peer-io.cc:465)
```

Reply:
15387e898c4f1bc80d0375cb8082d02ea9a8e814 Oh god thank you, that made me break a cold sweat.

I was going to change this from `set_seed()` to `set_upload_only()` and I thought I already did, apparently not.